yjit/src/codegen.rs

```diff
@@ -8006,6 +8006,12 @@ fn gen_send_dynamic<F: Fn(&mut Assembler) -> Opnd>(
     // Save PC and SP to prepare for dynamic dispatch
     jit_prepare_non_leaf_call(jit, asm);
 
+    // Squash stack canary that might be left over from elsewhere
+    assert_eq!(false, asm.get_leaf_ccall());
```

Comment:
I was wondering if we can do this in `jit_prepare_non_leaf_call()`. I think https://github.com/ruby/ruby/pull/10033 satisfied the assumption.